### PR TITLE
Clarify documentation for javafx:run

### PR DESCRIPTION
**Pull request by anthonyvdotbe**
-Xmx1024m is a valid VM option,
so it shouldn't be used as an example of a program argument

### CODE REVIEW

**Comment by abhinayagarwal** [comment]
Hi @anthonyvdotbe ,

Thank you for creating this PR. Can you please clarify what is the issue that you are trying to resolve?

**Comment by anthonyvdotbe** [comment]
Hi @abhinayagarwal

The way I understood:
> `commandlineArgs`: Arguments separated by space for the executed program

is that they're arguments that are passed to the main method of the application via its `String[]` parameter.

However, the example that follows:
```xml
<plugin>
    <groupId>org.openjfx</groupId>
    <artifactId>javafx-maven-plugin</artifactId>
    <version>0.0.4</version>
    <configuration>
        <mainClass>org.openjfx.hellofx/org.openjfx.App</mainClass>
        <options>
            <option>--add-opens</option>
            <option>java.base/java.lang=org.openjfx.hellofx</option>
        </options>
        <commandlineArgs>-Xmx1024m</commandlineArgs>
    </configuration>
</plugin>
```

suggests that `commandlineArgs` serves to pass VM options to the VM, since `-Xmx1024m` is a valid VM option.

So I'm just trying to clarify the documentation. If `commandlineArgs` is for main method arguments, then the example should be updated (as in this PR). If `commandlineArgs` is for VM options, then the initial description should be clarified to say `VM options` instead of `arguments`.

**Comment by abhinayagarwal** [comment]
In both `JavaFXRunMojo` and `JavaFXJlinkMojo`, `commandlineArgs` is passed to the actual `java` launcher instead of main method. Therefore, it seems that this is actually used for `VM options`.

**Comment by anthonyvdotbe** [comment]
> In both `JavaFXRunMojo` and `JavaFXJlinkMojo`, `commandlineArgs` is passed to the actual `java` launcher instead of main method. Therefore, it seems that this is actually used for `VM options`.

Actually they are passed to the main method. In [`JavaFXRunMojo`](https://github.com/openjfx/javafx-maven-plugin/blob/f325c425d90360a7a1a0f8f21540a75610956fac/src/main/java/org/openjfx/JavaFXRunMojo.java#L165), `commandlineArgs` is added *after* `mainClass`, which implies that they are passed to the main method.

Same for [`JavaFXJLinkMojo`](https://github.com/openjfx/javafx-maven-plugin/blob/4e4b7c4e3bf3552eb58bf2e805cd7baadad1c439/src/main/java/org/openjfx/JavaFXJLinkMojo.java#L276), where `commandlineArgs` is added by looking for the line in the launcher script that ends with `$@`.

Also note that the plugin already has `options` to configure VM options. So if `commandlineArgs` would also serve this purpose, there would actually be 2 ways to configure the same thing, and no way to configure main method arguments.

In conclusion, I believe this PR is correct as it is.

**Comment by abhinayagarwal** [comment]
Aha, good catch!

**Review by abhinayagarwal — APPROVED** [review]

**Comment by anthonyvdotbe** [comment]
@abhinayagarwal I just signed the Gluon CLA (which I had overlooked earlier). Is there anything else I should do before this can be merged?

File: README.md [hunk 1]
```diff
@@ -143,7 +143,7 @@ For instance, the following configuration adds some VM options and a command lin
             <option>--add-opens</option>
             <option>java.base/java.lang=org.openjfx.hellofx</option>
         </options>
-        <commandlineArgs>-Xmx1024m</commandlineArgs>
+        <commandlineArgs>foo</commandlineArgs>
```
Comment (jperedadnr) [inline README.md:146]: The proposed change is okay, but it would be good to show the new parameter in context (i.e. show also the `main` method that is expecting the `foo` option).

Comment (anthonyvdotbe) [reply README.md:146]: Just above this example, there's an overview of the different configuration options, including commandlineArgs:
> commandlineArgs: Arguments separated by space for the executed program

So I'd say that's sufficient context. If needed, that explanation could be expanded:
> These are passed to the main method via its String[] parameter (in the example below, `args.length == 1 && args[0].equals("foo")` would be `true`). To pass an argument that contains a space, ...

But I don't see the value-add of the first sentence, and don't know how to complete the second.

Comment (jperedadnr) [reply README.md:146]: I'd say, right after the code block add something like the following?
__________

so it can be processed by the `main` method like:

```
public static void main(String[] args) {
        if (args.length > 0 && "foo".equals(args[0])) {
             // do something
        }
        launch();
    }
```

Comment (anthonyvdotbe) [reply README.md:113]: Thanks for your suggestion. I've updated the PR accordingly.

**Review by johanvos — APPROVED** [review]

**Comment by johanvos** [comment]
Thanks @anthonyvdotbe  for this contribution!
